appendices/keywords.md

```diff
@@ -0,0 +1,60 @@
+# Keywords
+
+This listing explains the usage of every Pony keyword.
+
+|Keyword | Usage|
+| --- | --- |
+| actor | defines an actor
+| as | conversion of a value to another Type (can raise an error)
+| be | behavior, executed asynchronously
+| box | default reference capability – object is readable, but not writable
+| break | to step out of a loop statement
+| class | defines a class
+| compile_error | will provoke a compile error 
+| continue | loop statement
+| consume | move a value to a new variable, leaving the original variable empty
+| do | loop statement
+|    | after a with statement
+| else | conditional statement in if, for, while, repeat, try (as a catch block), match 
+| elseif | conditional statement, also used with ifdef
+| end | ending of if then, ifdef, while do, for in, repeat until, try, object, lambda,      |     | recover, match
+| error | raises an error
+| for | loop statement
+| fun | define a function, executed synchronously
+| if  | (1) conditional statement
+|     | (2) when specifying a guard in a pattern match
+| ifdef | when defining a build flag at compiling:  ponyc –D "foo"
+| in | loop statement
+| interface | used in structural subtyping
```

Comment:
The purpose of this list was not to give an exhaustive definition of each key word and its usage, but rather to indicate where the keyword is used.

In this particular example: the words structural subtyping should give enough hints to look for a complete definition, why and how it is used, etc.

 

 

 

From: Sean T Allen [mailto:notifications@github.com] 
Sent: woensdag 13 april 2016 15:00
To: ponylang/pony-tutorial <pony-tutorial@noreply.github.com>
Cc: ibalbaert <ivo.balbaert@telenet.be>
Subject: Re: [ponylang/pony-tutorial] adding keywords list (#98)

 

In appendices/keywords.md <https://github.com/ponylang/pony-tutorial/pull/98#discussion_r59542741> :

> +| compile_error | will provoke a compile error 
> +| continue | loop statement
> +| consume | move a value to a new variable, leaving the original variable empty
> +| do | loop statement
> +|    | after a with statement
> +| else | conditional statement in if, for, while, repeat, try (as a catch block), match 
> +| elseif | conditional statement, also used with ifdef
> +| end | ending of if then, ifdef, while do, for in, repeat until, try, object, lambda,      |     | recover, match
> +| error | raises an error
> +| for | loop statement
> +| fun | define a function, executed synchronously
> +| if  | (1) conditional statement
> +|     | (2) when specifying a guard in a pattern match
> +| ifdef | when defining a build flag at compiling:  ponyc –D "foo"
> +| in | loop statement
> +| interface | used in structural subtyping

used how would be my question

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub <https://github.com/ponylang/pony-tutorial/pull/98/files/7d41e7d6589471ae55f05e6d91184f2a8e8ee694#r59542741>   <https://github.com/notifications/beacon/AA6zfMPx1i_AvtwOKLO2ke3Rw1NOtV2rks5p3OjXgaJpZM4IGNqM.gif> 



---
Dit e-mailbericht is gecontroleerd op virussen met Avast antivirussoftware.
https://www.avast.com/antivirus